### PR TITLE
feat(payment): INT-2280 Added GooglePay for Checkout.com

### PR DESCRIPTION
**Pull request by HadesDX**
[INT-2280](https://jira.bigcommerce.com/browse/INT-2280)
## What?
Added googlepay for checkout.com
## Why?
to enable googlepay for checkout.com

## Testing / Proof
Screen Shot 2020-02-17 at 10 40 16 AM

@bigcommerce/checkout @bigcommerce/payments

### CODE REVIEW

File: src/customer/strategies/googlepay/googlepay-customer-mock.ts [hunk 1]
```diff
@@ -43,6 +43,30 @@ export function getBraintreeCustomerInitializeOptions(mode: Mode = Mode.Full): C
...
+        }
+        case Mode.UndefinedMethodId: {
+            return { ...undefinedMethodId, ...googlepayCheckoutcom };
+        }
+        case Mode.InvalidContainer: {
+            return { ...methodId, ...googlepayCheckoutcomWithInvalidContainer };
+        }
+        default: {
+            return { ...methodId, ...googlepayCheckoutcom };
+        }
```
Comment (mauricio-sg) [inline src/customer/strategies/googlepay/googlepay-customer-mock.ts:66]: What the curly braces are for on the cases and the default option?

File: src/payment/payment-methods.mock.ts [hunk 2]
```diff
@@ -444,6 +463,7 @@ export function getPaymentMethods(): PaymentMethod[] {
         getPaypalExpress(),
         getSquare(),
         getStripeV3(),
+        getCheckoutcom(),
```
Comment (mauricio-sg) [inline src/payment/payment-methods.mock.ts]: It seems that the list follows an alphabetical order so, I think it should be placed after getBraintreeVisaCheckout().

File: src/payment/strategies/googlepay/googlepay-checkoutcom-initializer.spec.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    describe('#teardown', () => {
+        let googlePayCheckoutcomInitializer: GooglePayCheckoutcomInitializer;
+
+        beforeEach(() => {
+            googlePayCheckoutcomInitializer = new GooglePayCheckoutcomInitializer();
+        });
+
+        it('teardowns the initializer', async () => {
+            await googlePayCheckoutcomInitializer.teardown().then(() => {
```
Comment (mauricio-sg) [inline src/payment/strategies/googlepay/googlepay-checkoutcom-initializer.spec.ts]: Why you use await and then at the same time?

File: src/payment/strategies/googlepay/googlepay-checkoutcom-initializer.spec.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        });
+
+        it('parses a response from google pay payload received', () => {
+            const tokenizePayload = googlePayCheckoutcomInitializer.parseResponse(getGooglePaymentCheckoutcomDataMock());
+
+            expect(tokenizePayload).toBeTruthy();
+            expect(tokenizePayload).toEqual(getGooglePayTokenizePayloadCheckoutcom());
+        });
+
+        it('throws when try to parse a response from google pay payload received', () => {
```
Comment (mauricio-sg) [inline src/payment/strategies/googlepay/googlepay-checkoutcom-initializer.spec.ts:59]:
```suggestion
        it('throws when try to parse a response from google pay payload received', () => {
            const response = googlePayCheckoutcomInitializer.parseResponse(getGooglePaymentDataMock());

            expect(response).toThrow(new InvalidArgumentError('Unable to parse response from Google Pay.'));
        });
```

**Review by mauricio-sg — COMMENTED** [review]
Hello, I have some doubts.

File: src/customer/strategies/googlepay/googlepay-customer-mock.ts (hunk 1 above)
Comment (HadesDX) [reply src/customer/strategies/googlepay/googlepay-customer-mock.ts:66]: it ensures isolated code block for definitions

File: src/payment/payment-methods.mock.ts (hunk 2 above)
Comment (HadesDX) [reply src/payment/payment-methods.mock.ts]: yes it does, ill move it

File: src/payment/strategies/googlepay/googlepay-checkoutcom-initializer.spec.ts (hunk 3 above)
Comment (HadesDX) [reply src/payment/strategies/googlepay/googlepay-checkoutcom-initializer.spec.ts]: i'll remove async/await, to just return

File: src/payment/strategies/googlepay/googlepay-checkoutcom-initializer.spec.ts (hunk 4 above)
Comment (HadesDX) [reply src/payment/strategies/googlepay/googlepay-checkoutcom-initializer.spec.ts:59]: i'll wrap the response into a function, as 'expect' requieres:
const response = () => { googlePayCheckoutcomInitializer.parseResponse(getGooglePaymentDataMock()); };

**Pull request closed without merging**